### DISCUSSION TITLE
Implementation of HashMaps?

### DISCUSSION THREAD

**Discussion by jschmidt92** [NikolaiF90/ArmaPersistentCoreSystem#5] [Ideas]
After looking through your very promising framework, I think that you may benefit using BI's [HashMap](https://community.bistudio.com/wiki/HashMap) System. I'm currently working on a mission/profile namespace save system with my PMC Framework for those who don't want to use DragonflyDB.

**Comment by NikolaiF90** [comment]
Thanks for the suggestion. I have incorporated this system in my other project recently. After months of works, I've also realized that this systems need a lot of works. I'm afraid to say that this system is still very poorly optimized though. I will update this system, once I've released my recent project.

**Reply by jschmidt92** [reply]
You could utilize `count` instead of `forEach` loop to help with some of the optimization, for example here's a snippet of code from your saveContainers function:

```sqf
{
    private _container = _x;
    private _containerData = [];
    
    _containerData pushBack ["class", typeOf _container];
    _containerData pushBack ["cargo", [_container] call F90_fnc_generateCargoData];
    _containerData pushBack ["posRotation", [_container] call F90_fnc_generatePositioningData];
        
    _containers pushBack _containerData;
   true
} count (Persistent_ContainersToSave);
```

**Comment by NikolaiF90** [comment]
Yes, I admit that this system has a very poor structure. I will improve it soon
